### PR TITLE
Add content-ID to REST API v2 entries endpoint

### LINKED ISSUE #5878: Add content ID to `EntriesResponse` in API v2
Content ID can be used to distinguish entities with the same name (same key) in different commits (branches, tags).

Having content ID in `EntriesResponse` can help to avoid an extra call to the `getContent` API.

### PR DESCRIPTION
**Pull request by snazy**
Fixes #5878

### CODE REVIEW

File: model/src/main/java/org/projectnessie/model/EntriesResponse.java [hunk 1]
```diff
@@ -50,8 +53,13 @@ static ImmutableEntry.Builder builder() {
...
     ContentKey getName();
 
-    static Entry entry(ContentKey name, Content.Type type) {
-      return ImmutableEntry.of(name, type);
+    @JsonView(Views.V2.class)
+    @Value.Parameter(order = 3)
+    @Nullable // for V1 backwards compatibility
+    String getContentId();
+
+    static Entry entry(ContentKey name, Content.Type type, String contentId) {
```
Comment (dimas-b) [inline model/src/main/java/org/projectnessie/model/EntriesResponse.java:61]: Would it make sense to keep the old override in case some old client rely on it? The new attribute is nullable, so we can maintain a compatible implementation of the old `entry` method.

File: servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/AbstractResteasyV1Test.java [hunk 2]
```diff
@@ -121,6 +122,17 @@ public void testBasic() {
...
+    EntriesResponse entries =
+        rest()
+            .get("trees/tree/{branch}/entries", newReference.getName())
+            .then()
+            .statusCode(200)
+            .extract()
+            .as(EntriesResponse.class);
+    assertThat(entries.getEntries())
+        .hasSize(1)
+        .allSatisfy(e -> assertThat(e.getContentId()).isNull());
```
Comment (dimas-b) [inline servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/AbstractResteasyV1Test.java:134]: Do you think a plain JSON string test like `testCommitMetaAttributes` for the new content ID attribute is worth adding (in addition to this test, or replacing this test)? My thinking is that java code blurs the case of having a `null` attribue and not having an attribute.

**Review by dimas-b — COMMENTED** [review]
Thanks for the quick fix. LGTM overall. Minor comments.

File: model/src/main/java/org/projectnessie/model/EntriesResponse.java (hunk 1 above)
Comment (snazy) [reply model/src/main/java/org/projectnessie/model/EntriesResponse.java:61]: not sure whether clients have a reason to call this method, but good point.

**Comment by snazy** [comment]
Hum - something's broken :(

**Comment by snazy** [comment]
Damn - if `namespaceDepth > 0` is specified, the "flattening" doesn't work ATM.

**Comment by snazy** [comment]
> Damn - if `namespaceDepth > 0` is specified, the "flattening" doesn't work ATM.

Ah - not an issue at all - that's only used from the namespace API, which doesn't get the content-ID anyway - so all good :)

**Comment by dimas-b** [comment]
> if namespaceDepth > 0 is specified [...]

`namespaceDepth` is not supported in API v2... Does this help?

**Review by dimas-b — APPROVED** [review]
